### PR TITLE
Patternlab/DP-21691 main nav

### PR DESCRIPTION
**Pull request by ygannett**
Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)

## Description

- Re-label `aria-label` in `<nav class="ma__header__hamburger__nav"` from **main navigation** to **main menu**  
- Add ` aria-label="Main navigation"` to `div.ma__header__hamburger__main-nav`.
- Remove the old mobile version main navigation snippets (search and utility nav) from the horizontal nav.
Drupal uses the templates in theming for the main nav. The change was done to be consistent between Mayflower and Drupal.  The Mayflower PR doesn't affect the rendering in Drupal.  So, no need to wait to merge Drupal PR before the Mayflower release contains this MF PR.{color}

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-21691)

## Steps to Test

1. Go to http://localhost:3000/patterns/05-pages-Homepage/05-pages-Homepage.html. Open inspect.
2. Find `nav.ma__header__hamburger__nav` has **aria-label="Main menu"**.
3. Find `div.ma__header__hamburger__main-nav` has **aria-label="Main navigation"**.
4. Find `div.ma__header__nav` has **aria-label="Main navigation"**.
5. Go to http://localhost:3000/patterns/05-pages-event/05-pages-event.html or any page but home page.
6. Find `nav.ma__header__hamburger__nav` has **aria-label="Main menu"**.
7. Find `div.ma__header__hamburger__main-nav` has **aria-label="Main navigation"**.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

*

#### @TODO

*

### CODE REVIEW

File: packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig [hunk 1]
```diff
@@ -69,14 +69,11 @@
     {% endif %}
   </div>
 
-  <nav class="ma__header__nav" aria-label="main navigation" id="main-navigation" role="navigation">
+  <nav class="ma__header__nav" aria-label="Main navigation" id="main-navigation" role="navigation">
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:72]: Should the the arial-label on `<nav>` be consistent with the mobile version (https://github.com/massgov/mayflower/pull/1401/files#diff-2cf6b391741fd8a18285aade0abe3524d313ec9ebc247bb96ec4c83e7ca08682R12)?

Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:72]:
```suggestion
  <nav class="ma__header__nav" aria-label="Main menu" id="main-navigation" role="navigation">

```

File: packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig [hunk 2]
```diff
@@ -69,14 +69,11 @@
     {% endif %}
   </div>
 
-  <nav class="ma__header__nav" aria-label="main navigation" id="main-navigation" role="navigation">
+  <nav class="ma__header__nav" aria-label="Main navigation" id="main-navigation" role="navigation">
     <div class="ma__header__nav-container">
       <div class="ma__header__main-nav">
```
Comment (clairesunstudio) [inline packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:74]:
```suggestion
      <div class="ma__header__main-nav" aria-label="Main navigation">

```

File: packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig (hunk 1 above)
Comment (ygannett) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:72]: L.12 is the blue bar at the top of the page(red block in the screenshot).  That includes the mobile component(orange block in the screenshot).  It contains the utility nav for desktop, the mobile component (menu button, the logo, the search, the main nav and the utility nav).  So, I put aria-label as "Main **menu**" to differentiate from the main nav component (topic links) itself.

L. 72 is the horizontal nav(green block in the screenshot).  So, it only contains the main nav component (topic links) but nothing else, so its aria-label is set as "Main **navigation**"

I think both are labeled as "Main navigation" would be more confusing in desktop display.

home-desktop

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:72]: Ok I think I understand where my confusion came from -- due to the HTML structure differences between desktop and mobile, the navigation role is assigned to two different things:
1. On desktop:
    - nav contains only the mainNav 
    - "Main navigation" is outside of "Main menu"
    - There are 2 nav items coexist
2. On mobile:
    - nav contains both the mainNav and utilityNav
    - "Main navigation" is inside of "Main menu"

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:72]: desktop: Screen Shot 2021-04-28 at 1 30 08 PM

mobile: Screen Shot 2021-04-28 at 1 29 37 PM

Comment (clairesunstudio) [reply packages/patternlab/styleguide/source/_patterns/03-organisms/by-template/header-mixed.twig:72]: I wonder if it makes sense to remove the nav role from "Main menu" and assign that on mobile to be on the "_main-nav" div or the "ul" element, in order to make the roles and labels consistent between mobile and desktop

**Comment by ygannett** [comment]
replaced with https://github.com/massgov/mayflower/pull/1407.

**Pull request closed without merging**